### ISSUE TITLE
Performance

### ISSUE THREAD

**Issue by macintoshpie** [urbanopt/modelica-builder#9]
Compared to raw text manipulations that were previously used in geojson-modelica, this is ~100x slower.

After some initial benchmarking it seemed like ~1/2 of the time was spent just parsing the files. It might be worth a bit of effort profiling/documenting performance.

Also might be worth checking out 'speedy-antlr-tool': https://github.com/amykyta3/speedy-antlr-tool

**Comment by macintoshpie** [comment]
Just ran some tests locally to get an idea of possible performance improvements.
Edited the example repo to use our modelica grammar, and ran the benchmark on the test_1.mo file from geojson-modelica-translator ([here](https://github.com/urbanopt/geojson-modelica-translator/blob/f779d945f4f1feec865dbcd08b94a0a2ea6e2ff3/tests/modelica/data/test_1.mo#L1)).

Results:
```
py_elapsed:  330.613 ms
cpp_elapsed: 55.355 ms
Speedup: 5.97
```

The cpp parser was about 6x faster than using the python parser (what we use currently).

Note that I didn't do any tests to verify the AST accuracy as that'd be a bit more involved work to get it fully integrated with our package.

**Comment by macintoshpie** [comment]
Ran some more comparisons after integrating speedy-antlr with the package (see #16 )

I used pytest-benchmark fixture, and had it parse a small file (the DCMotor.mo from test data) and a larger file (Office.mo) using the C++ and python parsers.

### Results
| name (time in ms) | min     | max      | mean     | stddev  | median   |
|-------------------|---------|----------|----------|---------|----------|
| python-small-file | 1.2806  | 96.4835  | 2.4554   | 7.9437  | 1.3993   |
| cpp-small-file    | 1.3121  | 111.0004 | 2.1051   | 6.5126  | 1.4461   |
| python-large-file | 64.2107 | 198.7874 | 136.4350 | 47.8739 | 150.0920 |
| cpp-large-file    | 65.2889 | 151.0811 | 111.1692 | 27.7759 | 114.6904 |

Using C++ parser was about 1.2x faster than the python parser.

**Comment by macintoshpie** [comment]
After further investigation, sparked by the strange reduction in performance from 5x to 1.2x, I finally managed to get the results I would expect. I think there were some issues with my virtual environment's installed package.

See the results here: https://gist.github.com/macintoshpie/3ee17ba3257be021252f6060f8b6e6bb

TLDR; C++ parser is 3-4x faster

**Comment by macintoshpie** [comment]
We've decided that using the speedy-antlr-tool is not necessary at this time. See the archive/speedy-antlr-tool tag for future reference on how to do this integration.
